documentation/design/user-stories/shared-infra.adoc

```diff
@@ -744,13 +778,96 @@ spec:
   allowOverride: true | false
 ```
 
-=== Configure scaling of the routers based on the connection load and number of messaging tenants
+=== Configure scaling of the routers based on the connection load and throughput
+
+Create/modify the `MessagingInfra` resource to include the `dynamic` strategy for the router. This strategy scales
+the number of router replicas based on the parameters `connections` and `messageThroughputSeconds` per router.
+
+* If the observed value of these parameters go above the max, the number of router replicas is increased.
+* If the observed value of these parameters go below min, the number of router replicas is decreased.
+
+```
+apiVersion: enmasse.io/v1beta2
+kind: MessagingInfra
+metadata:
+  name: myinfra
+spec:
+  router:
+    scalingStrategy:
+      dynamic:
+        # Specify the range of replicas this strategy can operate within
+        replicas:
+          min: 2
+          max: 5
+        # Strategy parameters that is used for determining if it should scale up/down
+        parameters:
+          # Number of connections _per router replica_
```

Comment:
How will this work?  Won't we need some mechanism to rebalance connections?  For instance I have connections min/max set to 10/1000 and a single router with 1001 connections.  The system spins up a second router which will have 0 connections.  A native algorithm could flip flop state, shutting down router 2 as it has too few and then spinning it up again.

Reply:
I think we would need to rebalance, otherwise scaling will not help as you state. I think if you have > 2 routers, we can do an approximation of that by doing a controlled rolling restart if we are above the minimum available required replicas. 

However, ideally I think the operator would terminate the existing connections on routers during a scaleup/down event.

Reply:
IdeallyI think  we'd want some kind of heuristic that could take into account the cost of the connection (sessions, link, message size/throughput) and feed that into a rebalancing algorithm.  That sounds tricky.  

I think ultimately we will want to do something with intelligence, but we don't necessarily need to include in 1.0.